components/style/values/specified/basic_shape.rs

```diff
@@ -120,11 +123,12 @@ impl InsetRect {
 }
 
 impl Parse for InsetRect {
-    fn parse(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
+    fn parse<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>)
+                     -> Result<Self, ParseError<'i>> {
         match input.try(|i| i.expect_function()) {
             Ok(ref s) if s.eq_ignore_ascii_case("inset") =>
                 input.parse_nested_block(|i| GenericInsetRect::parse_function_arguments(context, i)),
-            _ => Err(())
+            _ => Err(BasicParseError::ExpectedToken(Token::Function("inset".into())).into())
```

Comment:
Do you know if this `"inset".into()` will allocate or not? I'm assuming that since it's an static string it shouldn't need to, but I worry that perhaps to coerce it to do that you may need to be explicit about it. Not a big deal in any case I think